File: aemet/Aemet-Java-IvanRC/src/main/java/dto/RegistroDto.java
package dto;

import com.squareup.moshi.Json;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Json(name = "registro")
@Root(name = "registro")
public class RegistroDto {

    @Element(name = "dia")
    String dia;

    @Element(name = "temperatura_max", required = false)
    String temperaturaMax = null;

    @Element(name = "temperatura_min", required = false)
    String temperaturaMin = null;

    @Element(name = "temperatura_media", required = false)
    String temperaturaMedia = null;

    @Element(name = "lugar", required = false)
    String lugar = null;

    @Element(name = "momento", required = false)
    String momento = null;

    @Element(name = "hubo_precipitacion", required = false)
    String huboPrecipitacion = null;

    @Element(name = "precipitacion", required = false)
    String precipitacion = null;

    public RegistroDto(String dia, String temperaturaMedia) {
        this.dia = dia;
        this.temperaturaMedia = temperaturaMedia;
    }

    public RegistroDto(String dia, String temperaturaMax, String lugar, String momento) {
        this.dia = dia;
        this.temperaturaMax = temperaturaMax;
        this.lugar = lugar;
        this.momento = momento;
    }

    public RegistroDto(String dia, String temperaturaMin, String lugar, String momento, String campoVacioDiferenciador) {
        this.dia = dia;
        this.temperaturaMin = temperaturaMin;
        this.lugar = lugar;
        this.momento = momento;
    }

    public RegistroDto(String dia, String huboPrecipitacion, String precipitacion) {
        this.dia = dia;
        this.huboPrecipitacion = huboPrecipitacion;
        this.precipitacion = precipitacion;
    }
}

File: aemet/Aemet-kotlin-IvanRC/src/main/kotlin/dto/RegistroDto.kt
package dto

import com.squareup.moshi.Json
import org.simpleframework.xml.Element
import org.simpleframework.xml.Root

@Json(name = "registro")
@Root(name = "registro")
data class RegistroDto (
    @Json(name = "dia")
    @field:Element(name = "dia")
    @param:Element(name = "dia")
    val dia: String,

    @Json(name = "temperatura_max")
    @field:Element(name = "temperatura_max", required = false)
    @param:Element(name = "temperatura_max", required = false)
    val temperaturaMax: String? = null,

    @Json(name = "temperatura_min")
    @field:Element(name = "temperatura_min", required = false)
    @param:Element(name = "temperatura_min", required = false)
    val temperaturaMin: String? = null,

    @Json(name = "temperatura_media")
    @field:Element(name = "temperatura_media", required = false)
    @param:Element(name = "temperatura_media", required = false)
    val temperaturaMedia: String? = null,

    @Json(name = "lugar")
    @field:Element(name = "lugar", required = false)
    @param:Element(name = "lugar", required = false)
    val lugar: String? = null,

    @Json(name = "momento")
    @field:Element(name = "nombre", required = false)
    @param:Element(name = "nombre", required = false)
    val momento: String? = null,

    @Json(name = "hubo_precipitacion")
    @field:Element(name = "hubo_precipitacion", required = false)
    @param:Element(name = "hubo_precipitacion", required = false)
    val huboPrecipitacion: String? = null,

    @Json(name = "precipitacion")
    @field:Element(name = "precipitacion", required = false)
    @param:Element(name = "precipitacion", required = false)
    val precipitacion: String? = null
)